Add offloading page (#772)
MicroEJ/docs: VEEWear/offloading.rst

.. _offloading:

Offloading
==========

> Many modern smartwatches rely on a high-power microprocessor (such as a Qualcomm Snapdragon) running Android.

Author: fixed. (removed qualcomm reference)

> Integrating an additional low-power microcontroller to the hardware of the watch allows to offload a part of the work of the high-power microprocessor and thus increase the battery life.

Reviewer: ```suggestion
Integrating an extra low-power microcontroller into the watch's hardware enables the delegation of specific tasks from the main microprocessor, resulting in an increased battery life.
```

> Keeping a high-power microprocessor on the hardware allows to keep the ability to show high-performance animations and to provide access to the Android ecosystem.

Reviewer: two times "keep"
```suggestion
Keeping a powerful microprocessor on the hardware ensures the ability to display high-performance animations and access the Android ecosystem.
```

Solution
--------

> VEE Wear offers a comprehensive solution to develop the software on this dual architecture:

Reviewer: ```suggestion
VEE Wear offers a comprehensive solution  for software development on this dual architecture:
```

> - VEE allows to run an application programmed with high-level code and virtualization on the microcontroller

Reviewer: ```suggestion
- MicroEJ VEE enables the execution of applications written in high-level code through virtualization on the microcontroller.
```

> - the :ref:`Android Compatibility Kit <runnning_on_android>` allows to execute the same application on the microprocessor without programming it a second time

Reviewer: programming could be misleading with hardware programmation (flash).
```suggestion
- the :ref:`Android Compatibility Kit <runnning_on_android>` allows the execution of the same application on the microprocessor without the need for re-implementation.
```

> - the Offloading Framework allows to switch between the two processors depending on their capabilities and on the watch design

Reviewer: ```suggestion
- the Offloading Framework provides the ability to dynamically allocate processing tasks between the two processors depending on their capabilities and the application flow.
```

Reviewer: ok, maybe switch the application context between (...) ?

Author: fixed (with your last suggestion)

One Code, Two Targets
~~~~~~~~~~~~~~~~~~~~~

> MicroEJ Application development shares the same programming language as Android Application development.

Reviewer: Kotlin? :)

Thanks to this similarity, any MicroEJ Application code is compatible with the Android runtime environment.
The :ref:`Android Compatibility Kit <runnning_on_android>` provides the tools and libraries to execute a MicroEJ Application on Android.

Using this solution, the application code can be programmed once and executed both on the low-power microcontroller and the high-power microprocessor.

Offloading Framework
~~~~~~~~~~~~~~~~~~~~

The offloading strategy is a set of rules defined by the design of the watch.

Here is a non-exhaustive list of common offloading rules:

- switching to the low-power microcontroller after a few seconds of user inactivity (often called ambient mode)
- switching to the high-power microprocessor when starting a high-performance animation

> - switching to the high-power microprocessor when navigating to a menu which is only available on Android

Reviewer: e.g. third party app?

Author: it is a good example, but it could also be any Android menu which hasn't been ported to the VEE/MCU (settings menu or whatever)
